File: src/main/java/cc/duduhuo/git/conflict/model/PersistentState.java
package cc.duduhuo.git.conflict.model;

import java.util.*;

import cc.duduhuo.git.conflict.BuiltInColor;

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/25 21:22
 * Description:
 * Remarks:
 * =======================================================
 */
public final class PersistentState {
    private String mSchemeName;
    private LinkedHashMap<String, MarkColor> mMarkColors;

    public PersistentState() {
        mSchemeName = BuiltInColor.DEFAULT_SCHEME_NAME;
        mMarkColors = new LinkedHashMap<>();
        mMarkColors.put(BuiltInColor.AUTO_SCHEME_NAME, BuiltInColor.AUTO);
        mMarkColors.put(BuiltInColor.INTELLIJ_SCHEME_NAME, BuiltInColor.INTELLIJ);
        mMarkColors.put(BuiltInColor.DARCULA_SCHEME_NAME, BuiltInColor.DARCULA);
    }

    public PersistentState(String schemeName, LinkedHashMap<String, MarkColor> markColors) {
        this.mSchemeName = schemeName;
        this.mMarkColors = markColors;
    }

    public String getSchemeName() {
        return mSchemeName;
    }

    public void setSchemeName(String schemeName) {
        mSchemeName = schemeName;
    }

    public LinkedHashMap<String, MarkColor> getMarkColors() {
        return mMarkColors;
    }

    public void setMarkColors(LinkedHashMap<String, MarkColor> markColors) {
        mMarkColors = markColors;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PersistentState)) return false;
        PersistentState that = (PersistentState) o;
        return Objects.equals(mSchemeName, that.mSchemeName) &&
            Objects.equals(mMarkColors, that.mMarkColors);
    }

    @Override
    public int hashCode() {
        return Objects.hash(mSchemeName, mMarkColors);
    }
}

File: src/main/java/cc/duduhuo/git/conflict/model/PersistentState.kt
package cc.duduhuo.git.conflict.model

import cc.duduhuo.git.conflict.BuiltInColor

/**
 * =======================================================
 * Author: liying - liruoer2008@yeah.net
 * Datetime: 2018/4/25 21:22
 * Description:
 * Remarks:
 * =======================================================
 */
data class PersistentState(
    var schemeName: String = BuiltInColor.DEFAULT_SCHEME_NAME,
    var markColors: LinkedHashMap<String, PersistentMarkColor> = LinkedHashMap(
        mutableMapOf(
            BuiltInColor.DEFAULT_SCHEME_NAME to PersistentMarkColor.of(BuiltInColor.DEFAULT),
        )
    )
)
